nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-site-to-site/src/main/java/org/apache/nifi/remote/StandardRemoteGroupPort.java

```diff
@@ -366,6 +371,17 @@ private int receiveFlowFiles(final Transaction transaction, final ProcessContext
         return flowFilesReceived.size();
     }
 
+    private String createTransitUri(final Transaction transaction, String sourceFlowFileIdentifier) {
+        // Use actual uri as transitUri if this transport protocol has meaningful uri.
+        switch (remoteGroup.getTransportProtocol()) {
```

Comment:
This feels wrong to me. We should be using the URI that the communicant gives to us, regardless of which implementation is being used. If the URI returned by the HTTP implementation is not meaningful, then we should update the code so that we pass in a more meaningful URI